### PR TITLE
Add raw RPC calls

### LINKED ISSUE #987: Allow raw RPC calls via `cargo contract rpc`
```
cargo contract rpc [options] METHOD [PARAMS]
```

Users cannot make raw RPC calls to the node with the currently available commands.

Adding this capability would improve the debugging process by giving developers direct access to the node's RPC interface, making it easier to diagnose, fix, and experiment with different scenarios.

### PR DESCRIPTION
**Pull request by smiasojed**
## Summary
Closes #987
- [ ] y/n | Does it introduce breaking changes?
- [ ] y/n | Is it dependent on the specific version of `ink` or `pallet-contracts`?

Add raw RPC calls

## Description
Invoke an RPC call to the node in the format:
`cargo contract rpc [Options] METHOD [PARAMS]`

e.g.

```bash
cargo contract rpc author_insertKey '"sr25"' '"//ALICE"' \
      5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
```

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [x] I have added an entry to `CHANGELOG.md`
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: crates/cargo-contract/src/cmd/rpc.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2018-2024 Parity Technologies (UK) Ltd.
```
Comment (SkymanOne) [inline crates/cargo-contract/src/cmd/rpc.rs:1]:
```suggestion
// Copyright (C) Parity Technologies (UK) Ltd.
```

File: crates/extrinsics/src/rpc.rs [hunk 2]
```diff
@@ -0,0 +1,208 @@
+// Copyright 2018-2024 Parity Technologies (UK) Ltd.
```
Comment (SkymanOne) [inline crates/extrinsics/src/rpc.rs:1]:
```suggestion
// Copyright (C) Parity Technologies (UK) Ltd.
```

File: docs/rpc.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ cargo contract rpc author_hasKey
+     5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY '"sr25"' \
+```
+ 
+account can be provided as ss58 address:
+`5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY`
+or in hex:
+`0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d`
+
+Some of the commands require SS58 address in the string format:
```
Comment (SkymanOne) [inline docs/rpc.md:21]: Is there a way to automatically convert it to string to make it consistent? I see this as a source of user errors in future.

**Review by SkymanOne — APPROVED** [review]
LGTM, just few nits

File: docs/rpc.md (hunk 3 above)
Comment (smiasojed) [reply docs/rpc.md:21]: > Is there a way to automatically convert it to string to make it consistent? I see this as a source of user errors in future.

You would need to know for which command it is required. In this case, it means hardcoding the command name and matching agains it

Comment (SkymanOne) [reply docs/rpc.md:21]: Sure, then I would create an issue for this, so we can pick it up later

**Pull request merged**